syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.telemetry";
option java_outer_classname = "TelemetryProto";
option objc_class_prefix = "HLW";

service Telemetry {
  rpc SaveTemperature (Temperature) returns (Reply) {}
  rpc SaveHumidity (Humidity) returns (Reply) {}
  rpc SaveWeather (Weather) returns (Reply) {}
}

message Info {
  string timestamp = 1;
  int32 location_x = 2;
  int32 location_y = 3;
  string subscriber = 4;
}

message Temperature {
  Info info = 1;
  float temperature = 2;
}

message Humidity {
  Info info = 1;
  int32 humidity = 2;
}

message Weather {
  Info info = 1;
  enum WeatherType {
    SUN = 0;
    CLOUDS = 1;
    RAIN = 2;
    SNOW = 3;
    FOG = 4;
    WIND = 5;
  }
  repeated WeatherType weather_type = 2;
}

message Reply {
  int32 status = 1;
  string message = 2;
}